docs/design/managing-resources.md

```diff
@@ -85,3 +85,94 @@ metadata:
     name: DbClaim1
     uid: 038b49a6-e746-11e6-baf3-ee8d75af8f6e
 ```
+
+### smith.a.c/TprOutputNameModeField=`<NameModeField>`, smith.a.c/TprOutputNameSetterField=`<OutputNameSetterField>`, smith.a.c/TprOutputNameField=`<OutputNameField>`
+
+These two annotations work together with `smith.a.c/TprReadyWhenExistsKind` and `smith.a.c/TprReadyWhenExistsVersion`.
+Please refer to their definition for definitions of `Tinst` and `K`.
+
+`smith.a.c/TprOutputNameModeField` can be applied to a TPR `T` to configure the way name of the output object `O`
+is constructed. Object `O` is a [Secret](https://kubernetes.io/docs/user-guide/secrets/) or a
+[ConfigMap](https://kubernetes.io/docs/user-guide/configmap/) which holds parameters that may be injected into
+object(s) `D` depending on `Tinst`. Object `O` is created/updated before `K` is created/updated.
+
+The following values of `smith.a.c/TprOutputNameModeField` are defined:
+- `Random` - `O`'s name is randomly generated on each `O` update.
+- `ContentsHash` - `O`'s name is a deterministic hash function of `O`'s contents. That means name changes iif contents change.
+- `Fixed` - `O`'s name is provided via a field on `Tinst`. Name of that field is defined by
+`smith.a.c/TprOutputNameSetterField` set on `T`.
+
+Each time `O`'s name changes `O` object is recreated, old object is deleted immediately and all dependent objects
+will have references to `O` updated.
```

Comment:
I'm assuming we will want to create a new object, update references, and afterwards remove the old object? If so, maybe:

> Each time `O`'s name changes a copy `O` object is created, all dependent objects will have references updated `O` and the old object will be deleted.